src/org/skypro/skyshop/Search/SearchEngine.java

```diff
@@ -1,31 +1,35 @@
 package org.skypro.skyshop.Search; // ПОИСК это серч
 
+import org.skypro.skyshop.MaxLengthNameComparator;
 import org.skypro.skyshop.application.errors.BestResultNotFound;
-import org.skypro.skyshop.product.Product;
 
 import java.util.*;
 
 public class SearchEngine { // Поисковый движок
-    private Map<String, List<Searchable>> searchable; // заменила на Map
+    private Set<Searchable> searchable; // заменила
 
     public SearchEngine() {
-        this.searchable = new HashMap<>(); // поменяла конструктор на TreeMap
+        this.searchable = new HashSet<>(); // поменяла конструктор
     }
 
-    public void add(Searchable searchableName) { // все созданные элементы (продукты и статьи) добавляются в searchable
-        List searchableList = searchable.computeIfAbsent(searchableName.gettingSearchTerm(), k -> new ArrayList<>()); // лист для значений, если нет, будет создан
-        searchableList.add(searchableName); // положен термин в лист значений
-        searchable.put(searchableName.gettingSearchTerm(), searchableList); // добавление ключа и значения
-        System.out.println("мапа содержит теперь  = " + searchable);
+    public void add(Searchable searchableName) { // все созданные элементы добавляются в мно searchable
+        searchable.add(searchableName);
     }
-    public Map<String, ArrayList> search(String searchableName) {
-        System.out.println("searchable.get(searchableName)" + searchable.get(searchableName));
-        Map<String, ArrayList> result = new TreeMap<>(); // создала новую мапу для найденных ЭЛЕМЕНТОВ(статьи и продукты)
-        for (String name : searchable.keySet()) { // идем по всем ключам
-            if (name.equals(searchableName)) { // если строковые поля равны, ключ и поисковое слово
-                result.put(name, (ArrayList) searchable.get(searchableName)); // добавляем в ключ этот ключ, а в значение значение этого поискового слова
+
+    public Set<Searchable> search(String searchableName) {
+        Set<Searchable> result = new TreeSet<>(new MaxLengthNameComparator()); // добавила компаратор здесь, там сортировка по длине имени метода getNameLength()
+        Iterator<Searchable> iterator = searchable.iterator();// получаем итератор по сету searchable
+        System.out.println("Идет поиск по запросу - " + searchableName);
+        while (iterator.hasNext()) { // пока есть след/элемент в сете
```

Comment:
Только один комментарий - итерироваться можно с помощью цикла for, без использование итератора
```suggestion
        for (Searchable element: searchable)
```